packages/devextreme/js/__internal/ui/date_range_box/m_date_range_box.ts

```diff
@@ -41,1003 +47,1081 @@ const DROP_DOWN_EDITOR_ACTIVE_CLASS = 'dx-dropdowneditor-active';
-            onCut: null,
-            onEnterKey: null,
-            onInput: null,
-            onKeyDown: null,
-            onKeyUp: null,
-            onOpened: null,
-            onPaste: null,
-            onValueChanged: null,
-            openOnFieldClick: true,
-            opened: false,
-            pickerType: 'calendar',
-            readOnly: false,
-            showClearButton: false,
-            showDropDownButton: true,
-            spellcheck: false,
-            startDate: null,
-            startDateInputAttr: {},
-            startDateLabel: messageLocalization.format('dxDateRangeBox-startDateLabel'),
-            startDateName: '',
-            startDateOutOfRangeMessage: messageLocalization.format('dxDateRangeBox-startDateOutOfRangeMessage'),
-            startDatePlaceholder: '',
-            startDateText: undefined,
-            stylingMode: config().editorStylingMode || 'outlined',
-            todayButtonText: messageLocalization.format('dxCalendar-todayButtonText'),
-            useHiddenSubmitElement: false,
-            useMaskBehavior: false,
-            validationError: null,
-            validationErrors: null,
-            validationMessageMode: 'auto',
-            validationMessagePosition: 'auto',
-            validationStatus: 'valid',
-            value: [null, null],
-            valueChangeEvent: 'change',
-            _internalValidationErrors: [],
-            _currentSelection: 'startDate',
-        });
-    }
-
-    _defaultOptionsRules() {
-        return super._defaultOptionsRules().concat([
-            {
-                device: function() {
-                    const themeName = current();
-                    return isMaterial(themeName);
-                },
-                options: {
-                    labelMode: 'floating',
-                    stylingMode: config().editorStylingMode || 'filled',
-                }
-            },
-            {
-                device: function() {
-                    const themeName = current();
-                    return isFluent(themeName);
-                },
-                options: {
-                    labelMode: 'outside'
-                }
-            },
-            {
-                device: function() {
-                    const realDevice = devices.real();
-                    const platform = realDevice.platform;
-                    return platform === 'ios' || platform === 'android';
-                },
-                options: {
-                    multiView: false
-                }
-            },
-        ]);
-    }
+class DateRangeBox extends Editor<Properties> {
+  private _openAction;
+
+  private _closeAction;
+
+  private _startDateBox;
+
+  private _endDateBox;
+
+  private _$startDateBox;
+
+  private _$endDateBox;
+
+  private _$separator;
+
+  private _popupContentId;
+
+  private _buttonCollection;
+
+  private _shouldSkipIsValidChange;
+
+  public _$beforeButtonsContainer;
+
+  public _$afterButtonsContainer;
```

Comment:
let's write `any` explicitly, e.g. `private _openAction: any`
It will help us to find all places with untyped properties easier in future